Source/Cogworks.Essentials/Services/MemoryCacheService.cs

```diff
@@ -1,30 +1,72 @@
-﻿using System;
-using Microsoft.Extensions.Caching.Memory;
+using System;
+using System.Linq;
+using System.Runtime.Caching;
+using Cogworks.Essentials.Constants;
+using Cogworks.Essentials.Services.Interfaces;
 
 namespace Cogworks.Essentials.Services
 {
-    public class MemoryCacheService : ICacheService
+    public class MemoryCacheService : IMemoryCacheService
```

Comment:
IMHO we should go with 1 of those:
1) extend existing object with functions: `public T GetOrAddValue<T>(string cacheKey, Func<T> getValueFunction, int? cacheDuration)`, `public void DeleteAllStartingWith(string key)`, `public void DeleteAll()` and revert back `ICache` and perform one of those:
 a) 
```
public interface IMemoryCacheService : ICacheService
```

or 
b) 
```
public class MemoryCacheService : ICacheServicem IMemoryCacheService
```

2) Creating separate `MemoryCacheService` lets call it f.e. `ProxyMemoryCacheService` that implement this logic

3) Abstract class with previous approach (as virtual) and extend it in this with some logic that works on those methods in abstract f.e. like `GetOrAdd<T>`

The previous approach is different as this, in the previous we are creating generic approach of handling stores in cache -> we need provide build key itself and we have separate methods for adding, getting and checking item in cache. We could extend it with some different like clear cache. We could also create abstract class with some default logic that can be used in some methods.

Happy to discuss it with all :)